pandas/core/arrays/sparse/array.py

```diff
@@ -766,7 +769,9 @@ def __getitem__(self, key):
                 else:
                     key = np.asarray(key)
 
-            if com.is_bool_indexer(key) and len(self) == len(key):
+            if com.is_bool_indexer(key):
+                key = check_bool_indexer(self, key)
```

Comment:
why is this check_bool_indexer when in all the others its check_bool_array_indexer?  found this bc mypy is complaining about that the first arg should be an Index